### ISSUE TITLE
Output template %(resolution)s and %(format_id)s lost when merging formats

### ISSUE THREAD

**Issue by newzealandpaul** [rg3/youtube-dl#3634]
If an output template contains %(resolution)s and %(format_id)s these variables are replaced with "NA" rather than the correct values if the format is a merge of a video and audio source

For example running the following command ...

youtube-dl -f 137+140  -o "%(title)s %(format_id)s %(resolution)s.%(ext)s"  https://www.youtube.com/watch?v=VPRjCeoBqrI

produces a file named ...

Coldplay - A Sky Full Of Stars (Official video) NA NA.mp4

I think the expect behavior would be a file named:

Coldplay - A Sky Full Of Stars (Official video) 1080p 137+140.mp4

**Comment by endumiuz** [comment]
Any news about this bug? Is it getting fixed?

**Comment by endumiuz** [comment]
Looks like the %(resolution)s part have been fixed
```
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['--verbose', '-f', 'bestvideo+bestaudio', '-o', '%(title)s %(format_id)s %(resolution)s.%(ext)s', 'http://youtu.be/l18A5BOTlzE']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2015.01.25
[debug] Python version 2.7.8 - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-69233-g1f13348
[debug] Proxy map: {}
[youtube] l18A5BOTlzE: Downloading webpage
[youtube] l18A5BOTlzE: Extracting video information
[youtube] l18A5BOTlzE: Downloading DASH manifest
[debug] Invoking downloader on 'https://r4---sn-p5qlsm76.googlevideo.com/videoplayback?id=975f00e413939731&itag=137&source=youtube&requiressl=yes&mm=31&nh=EAE&pl=19&mv=m&ms=au&ratebypass=yes&mime=video/mp4&gir=yes&clen=61009849&lmt=1410401309260946&d
ur=177.777&upn=1gy2nlcYV7s&key=dg_yt0&signature=8B36F249EC69EC6B87D3A0589C55129510466CE8.13FBF59C499CC214DD00A0CFD6D9D62F849104DF&sver=3&fexp=900228,900718,902404,907263,927622,930676,9405761,9406627,942504,943917,946008,947225,948124,948803,952302,9
52605,952901,955301,957201,959701&mt=1422469758&ip=194.117.188.224&ipbits=0&expire=1422491443&sparams=ip,ipbits,expire,id,itag,source,requiressl,mm,nh,pl,mv,ms,ratebypass,mime,gir,clen,lmt,dur'
[download] Destination: The Living Tombstone - Five Nights at Freddy's Song! 137 1920x1080.f137.mp4
[download] 100% of 58.18MiB in 00:15
[debug] Invoking downloader on 'https://r4---sn-p5qlsm76.googlevideo.com/videoplayback?id=975f00e413939731&itag=141&source=youtube&requiressl=yes&mm=31&nh=EAE&pl=19&mv=m&ms=au&ratebypass=yes&mime=audio/mp4&gir=yes&clen=5671335&lmt=1410401181352216&du
r=177.864&upn=1gy2nlcYV7s&key=dg_yt0&signature=3EE95A2CDC0BA0669B94582523D670BFDDAB852B.6107513F1392083C0E14AA5F4BBE62848AE2E426&sver=3&fexp=900228,900718,902404,907263,927622,930676,9405761,9406627,942504,943917,946008,947225,948124,948803,952302,95
2605,952901,955301,957201,959701&mt=1422469758&ip=194.117.188.224&ipbits=0&expire=1422491443&sparams=ip,ipbits,expire,id,itag,source,requiressl,mm,nh,pl,mv,ms,ratebypass,mime,gir,clen,lmt,dur'
[download] Destination: The Living Tombstone - Five Nights at Freddy's Song! 141 NA.f141.m4a
[download] 100% of 5.41MiB in 00:05
[ffmpeg] Merging formats into "The Living Tombstone - Five Nights at Freddy's Song! NA 1920x1080.mp4"
[debug] ffmpeg command line: ffmpeg -y -i 'The Living Tombstone - Five Nights at Freddy'"'"'s Song! 137 1920x1080.f137.mp4' -i 'The Living Tombstone - Five Nights at Freddy'"'"'s Song! 141 NA.f141.m4a' -c copy -map 0:v:0 -map 1:a:0 'The Living Tombst
one - Five Nights at Freddy'"'"'s Song! NA 1920x1080.mp4'
```

**Comment by dstftw** [comment]
Will be fixed in the next version.

**Comment by phihag** [comment]
I believe this issue to be fixed in youtube-dl version 2015.01.30.1 and newer. See [our FAQ](https://yt-dl.org/update) if you need help updating.

**Comment by newzealandpaul** [comment]
Confirmed fixed. Thanks guys for all the hard work you put into this amazing project.
